### PR TITLE
SRV-291 Fetch exchange rate (mock)

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

File: app/Client/DummyExchangeRateExternalProvider.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+declare(strict_types = 1);
+
+namespace Adshares\Adserver\Client;
+
+use Adshares\Adserver\Utilities\DateUtils;
+use Adshares\Common\Application\Dto\FetchedExchangeRate;
+use Adshares\Common\Application\Service\ExchangeRateExternalProvider;
+use DateTime;
+
+class DummyExchangeRateExternalProvider implements ExchangeRateExternalProvider
```
Comment (teghnet) [inline app/Client/DummyExchangeRateExternalProvider.php]:
```suggestion
class DummyExchangeRateExternalProvider implements ExchangeRateProvider
```

File: app/Repository/Common/ExchangeRateRepositoryImpl.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace Adshares\Adserver\Repository\Common;
+
+use Adshares\Adserver\Models\ExchangeRate;
+use Adshares\Common\Application\Dto\FetchedExchangeRate;
+use Adshares\Common\Application\Service\Exception\ExchangeRateNotAvailableException;
+use Adshares\Common\Application\Service\ExchangeRateRepository;
+use DateTime;
+
+class ExchangeRateRepositoryImpl implements ExchangeRateRepository
```
Comment (teghnet) [inline app/Repository/Common/ExchangeRateRepositoryImpl.php]:
```suggestion
class EloquentExchangeRateRepository implements ExchangeRateRepository
```

File: database/migrations/2019_04_08_142523_create_exchange_rates_table.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+class CreateExchangeRatesTable extends Migration
+{
+    public function up(): void
+    {
+        Schema::create(
+            'exchange_rates',
+            function (Blueprint $table) {
+                $table->increments('id');
+                $table->timestamps();
+                $table->timestamp('valid_at')->nullable(false)->default('1970-01-01 00:00:01');
```
Comment (teghnet) [inline database/migrations/2019_04_08_142523_create_exchange_rates_table.php]:
```suggestion
                $table->timestamp('valid_at')->nullable(false)->unique();
```

File: src/Common/Application/Dto/FetchedExchangeRate.php [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    /** @var DateTime */
+    private $dateTime;
+
+    /** @var string */
+    private $value;
+
+    /** @var string */
+    private $currency;
+
+    public function __construct(DateTime $dateTime, string $value, string $currency)
```
Comment (teghnet) [inline src/Common/Application/Dto/FetchedExchangeRate.php]:
```suggestion
    public function __construct(DateTimeInterface $dateTime, string $value, string $currency)
```

File: app/Repository/Common/ExchangeRateRepositoryImpl.php (hunk 2 above)
Comment (PawelPodkalicki) [reply app/Repository/Common/ExchangeRateRepositoryImpl.php]: I'll rename

File: app/Client/DummyExchangeRateExternalProvider.php (hunk 1 above)
Comment (PawelPodkalicki) [reply app/Client/DummyExchangeRateExternalProvider.php]: Renamed

File: database/migrations/2019_04_08_142523_create_exchange_rates_table.php (hunk 3 above)
Comment (PawelPodkalicki) [reply database/migrations/2019_04_08_142523_create_exchange_rates_table.php]: Solved by moving timestamps at the end of columns definitions

**Review by c3zi — APPROVED** [review]

**Pull request merged**